gateway/envoy.yaml

```diff
@@ -0,0 +1,146 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: gateway-envoy-config
+data:
+  envoy.yaml: |
+    static_resources:
+      listeners:
+      - address:
+          socket_address:
+            address: 0.0.0.0
+            port_value: 80
+        filter_chains:
+        - filters:
+          - name: envoy.filters.network.http_connection_manager
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+              codec_type: AUTO
+              stat_prefix: ingress_http
+              route_config:
+                name: http_routes
+                virtual_hosts:
+                  - name: revoke
+                    domains:
+                      - "bogus-for-revoke.*"
+                    routes: []
+                  - name: default_http
+                    domains: ["*"]
+                    routes:
+                      - match:
+                          prefix: "/.well-known/acme-challenge"
+                        route:
+                          timeout: 0s
+                          cluster: letsencrypt
+                      - match:
+                          prefix: "/"
+                        redirect:
+                          https_redirect: true
+              http_filters:
+                - name: envoy.filters.http.router
+                  typed_config:
+                    "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
+      - address:
+          socket_address:
+            address: 0.0.0.0
+            port_value: 443
+        filter_chains:
+        - filters:
+          - name: envoy.filters.network.http_connection_manager
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+              codec_type: AUTO
+              stat_prefix: ingress_http
+              upgrade_configs:
+                - upgrade_type: websocket
+              use_remote_address: true
+              skip_xff_append: false
+              rds:
+                route_config_name: https_routes
+                config_source:
+                  resource_api_version: V3
+                  path_config_source:
+                    path: /config_map/xds/rds.yaml
+                    watched_directory:
+                      path: /config_map/xds
+              http_filters:
+                - name: envoy.filters.http.ext_authz
+                  typed_config:
+                    "@type": type.googleapis.com/envoy.extensions.filters.http.ext_authz.v3.ExtAuthz
+                    http_service:
+                      server_uri:
+                        uri: https://auth:443
+                        cluster: auth
+                        timeout: 0.25s
+                      path_prefix: /api/v1alpha/verify_dev_credentials?url=
```

Comment:
Same comment as internal-gateway about providing a simpler example or documentation on how this is all structured.

Reply:
Here's an example from their repo on using the [ext_authz](https://github.com/envoyproxy/envoy/blob/main/examples/ext_authz/config/http-service.yaml) filter.